docs/source/documentation/diagrams/architecture-diagrams.html.md.erb

```diff
@@ -0,0 +1,33 @@
+# Data Platform - Architectural Diagrams
```

Comment:
No dash needed and sentence case for titles: Data Platform architectural diagrams

Reply:
I'd include a sentence here that outlines the diagrams this page covers, as an overview.

Reply:
Is it worth adding in a sentence here to introduce this pages as an index of architectural diagrams? I realise it only has the C4 ones so far, which are described below, but worth including if you add more diagrams to this page.